test/test_cases/run_3d_rising_bubble.jl

```diff
@@ -35,32 +35,196 @@ function run_3d_rising_bubble(
+    )
 
     # execute differently depending on testing mode
-    if test_mode == :regression
+    if mode == :integration
         # TODO!: run with input callbacks = ...
-        simulation = Simulation(model, stepper, dt = dt, tspan = (0.0, 1.0))
-        @test simulation isa Simulation
+        # Runs model with all thermodynamic configurations to ensure that the 
+        # assembled model is sensible, and that variable definitions and 
+        # timestepper progress is satisfactory. 
+        @testset "Integration: Potential Temperature Model" begin
+            simulation =
+                Simulation(model_pottemp, stepper, dt = dt, tspan = (0.0, 1.0))
+            @test simulation isa Simulation
+            # test set function
+            @unpack ρ, uh, w, ρθ = init_dry_rising_bubble_3d(FT, params, :ρθ)
+            set!(simulation, :base, ρ = ρ, uh = uh, w = w)
+            set!(simulation, :thermodynamics, ρθ = ρθ)
+            # test error handling
+            @test_throws ArgumentError set!(simulation, quack = ρ)
+            @test_throws ArgumentError set!(simulation, ρ = "quack")
+            # test successful integration
+            @test step!(simulation) isa Nothing # either error or integration runs
+        end
+        @testset "Integration: Total Energy Model" begin
+            simulation = Simulation(model, stepper, dt = dt, tspan = (0.0, 1.0))
+            @test simulation isa Simulation
+            # test set function
+            @unpack ρ, uh, w, ρe_tot =
+                init_dry_rising_bubble_3d(FT, params, :ρe_tot)
+            set!(simulation, :base, ρ = ρ, uh = uh, w = w)
+            set!(simulation, :thermodynamics, ρe_tot = ρe_tot)
+            # test error handling
+            @test_throws ArgumentError set!(simulation, quack = ρ)
+            @test_throws ArgumentError set!(simulation, ρ = "quack")
+            # test successful integration
+            @test step!(simulation) isa Nothing # either error or integration runs
+        end
+    elseif mode == :regression
+        @testset begin
+            # Regression checks compare current acceptable standard of variables
+            # ρ, θ, [TODO: and in the case of moisture: ρq_tot], given a 
+            # pre-determined test resolution, and pre-determined test end-time. 
+            # Conservation of key prognostic variables is also verified. Upon
+            # the addition of new model components these tests can be re-run to 
+            # ensure solution stability is unaffected, and that deviations from
+            # current variable bounds is satisfactory. 
+            #
+            # For the RTB test, the typical test time is 700s. 
+            # TODO: Fix when time-to-solution is reasonable. 
+            "Regression: Potential Temperature Model"
+            simulation =
+                Simulation(model_pottemp, stepper, dt = dt, tspan = (0.0, 1.0))
+            @unpack ρ, uh, w, ρθ = init_dry_rising_bubble_3d(FT, params, :ρθ)
```

Comment:
function name has changed

Reply:
Noted, now uses Utils and updates fn name.